### PR TITLE
Implement `FlowBase.__eq__` to allow comparing two flows

### LINKED ISSUE #101: Override '__eq__' method of FlowBase to allow compare two flows
Today, FlowBase class in flow.py module doesn't offer an easy way to compare two flows. The override of method '__eq__' it is a good solution to this problem.

### PR DESCRIPTION
**Pull request by cmagnobarbosa**
### :octocat: Are you working on some issue? Identify the issue!

Fix #101 

### :bookmark_tabs: Description of the Change

Override ``__eq__`` of ``FlowBase`` Class to compare two ``flows``.

### :page_facing_up: Release Notes

- Added support to compare two flows in FlowBase Class.

### CODE REVIEW

File: flow.py [hunk 1]
```diff
@@ -209,6 +209,16 @@ def from_of_flow_stats(cls, of_flow_stats, switch):
                    actions=non_none_actions,
                    stats=FlowStats.from_of_flow_stats(of_flow_stats))
 
+    def __eq__(self, other):
+        include_id = False
+        if not isinstance(other, self.__class__):
+            raise ValueError(f'Error comparing two flows, the {other}'
+                             'is not a FlowBase class')
```
Comment (hdiogenes) [inline flow.py:216]: There's a wrong assumption here that `self.__class__` will always be `FlowBase`. It'll be whatever class inherited from FlowBase (eg.: `v0x01.Flow`). Either you change the comparison to check just for FlowBase (and allow for comparing different classes of Flow, like comparing a v0x01 Flow against a v0x04) or you change the message to make it explicit that you won't allow comparing different versions of Flow (which I think makes more sense and suggested below). 

```suggestion
        if not isinstance(other, self.__class__):
            raise ValueError(f'Error comparing flows: {other} is not '
                             f'an instance of {self.__class__}')
```

File: flow.py [hunk 2]
```diff
@@ -209,6 +209,16 @@ def from_of_flow_stats(cls, of_flow_stats, switch):
...
+
+        if self.as_dict(include_id) == other.as_dict(include_id):
+            return True
+        return False
```
Comment (hdiogenes) [inline flow.py:220]:
```suggestion
        return self.as_dict(include_id=False) == other.as_dict(include_id=False)
```

File: flow.py [hunk 3]
```diff
@@ -209,6 +209,16 @@ def from_of_flow_stats(cls, of_flow_stats, switch):
                    actions=non_none_actions,
                    stats=FlowStats.from_of_flow_stats(of_flow_stats))
 
+    def __eq__(self, other):
+        include_id = False
```
Comment (hdiogenes) [inline flow.py:213]:
```suggestion
```

**Review by hdiogenes — CHANGES_REQUESTED** [review]
I think this is the kind of change that begs for an unit test.

**Review by ajoaoff — APPROVED** [review]

**Comment by rmotitsuki** [comment]
One suggestion, add an inequality unit test just to make sure it's working too.

File: setup.cfg [hunk 4]
```diff
@@ -3,15 +3,16 @@ exclude = .eggs,ENV,build,docs/conf.py,venv
 
 [yala]
 radon mi args = --min C
-pylint args = --disable=too-few-public-methods,too-many-instance-attributes --ignored-modules=napps.kytos.mef_eline
+pylint args = --disable=too-few-public-methods,too-many-instance-attributes,unnecessary-comprehension
+--ignored-modules=napps.kytos.of_core
```
Comment (hdiogenes) [inline setup.cfg:7]: Does this really work as expected in a new line?

**Comment by josemauro** [comment]
@rmotitsuki we already have a test called "test__eq__success_with_different_flows" that tests an inequality. 
Also, we have a "test__eq__fail" that tests the case where "__eq__" receives objects with different types.

**Comment by rmotitsuki** [comment]
> @rmotitsuki we already have a test called "test__eq__success_with_different_flows" that tests an inequality.
> 
> https://github.com/kytos/of_core/blob/5536d7ff381c71462b5d4b9de754eb9ffc387fdc/tests/unit/test_flow.py#L130
> 
> 
> Also, we have a "test__eq__fail" that tests the case where "**eq**" receives objects with different types.
> https://github.com/kytos/of_core/blob/5536d7ff381c71462b5d4b9de754eb9ffc387fdc/tests/unit/test_flow.py#L171

@josemauro You test if the equality operator is False.
Inequality operator is the " != ". It should be fine in Python3, but is just another unit test to make sure everything is working.

**Pull request merged**